behave_django/environment.py

```diff
@@ -17,18 +37,15 @@ def before_scenario(self, context):
 
         Sets up the test case, base_url, and the get_url() utility function.
         """
+        context.__class__ = PatchedContext
+        # Simply setting __class__ directly doesn't work
+        # because behave.runner.Context.__setattr__ is implemented wrongly.
+        object.__setattr__(context, '__class__', PatchedContext)
```

Comment:
Regarding simple monkey patching: that's what I tried to do first, but found it difficult to patch a `@property`.